src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -44,20 +44,31 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             }
             error = null;
 #if CORECLR
-            object obj = JsonConvert.DeserializeObject(input, new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.None, MaxDepth = 1024 });
-
-            // JObject is a IDictionary
-            if (obj is JObject)
+            object obj = null;
+            try
             {
+                obj = JsonConvert.DeserializeObject(input, new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.None, MaxDepth = 1024 });
+
+                // JObject is a IDictionary
                 var dictionary = obj as JObject;
-                obj = PopulateFromJDictionary(dictionary, out error);
+                if (dictionary != null)
+                {
+                    obj = PopulateFromJDictionary(dictionary, out error);
+                }
+                else
+                {
+                    // JArray is a collection
+                    var list = obj as JArray;
+                    if (list != null)
+                    {
+                        obj = PopulateFromJArray(list, out error);
+                    }
+                }
             }
-
-            // JArray is a collection
-            else if (obj is JArray)
+            catch (JsonException je)
             {
-                var list = obj as JArray;
-                obj = PopulateFromJArray(list, out error);
+                // the same as JavaScriptSerializer does
+                throw new ArgumentException("input", je);
```

Comment:
> new ArgumentException("input", je);

The first parameter is supposed to be the exception message. You need to add a resource string in `WebCmdletStrings.resx` for the message. The string could be something like this: `Conversion from JSON failed with error: {0}`. So the message generated would be `var message = string.Format(CurrentCutlure, WebCmdletStrings.ResourceStringName, je.Message)`.